src/mc_tasks/ImpedanceTask.cpp

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2015-2020 CNRS-UM LIRMM, CNRS-AIST JRL
+ */
+
+#include <mc_rbdyn/configuration_io.h>
+#include <mc_tasks/ImpedanceTask.h>
+#include <mc_tasks/MetaTaskLoader.h>
+
+#include <mc_rtc/gui/ArrayLabel.h>
+#include <mc_rtc/gui/Transform.h>
+
+namespace mc_tasks
+{
+
+namespace force
+{
+
+ImpedanceTask::ImpedanceTask(const std::string & surfaceName,
+                             const mc_rbdyn::Robots & robots,
+                             unsigned int robotIndex,
+                             double stiffness,
+                             double weight,
+                             double impM,
+                             double impD,
+                             double impK)
+: SurfaceTransformTask(surfaceName, robots, robotIndex, stiffness, weight)
+{
+  const auto & robot = robots.robot(robotIndex);
+  if(!robot.surfaceHasIndirectForceSensor(surfaceName))
+  {
+    mc_rtc::log::error_and_throw<std::runtime_error>(
+        "[mc_tasks::ImpedanceTask] Surface {} does not have a force sensor attached", surfaceName);
+  }
+
+  type_ = "impedance";
+  name_ = "impedance_" + robots.robot(rIndex).name() + "_" + surfaceName;
```

Comment:
It's a bit nicer to use the more complete task name in the error here (it can sometimes be a bit tricky to track back where exception arose, so the more information the better):
```suggestion
  type_ = "impedance";
  name_ = "impedance_" + robots.robot(rIndex).name() + "_" + surfaceName;
  
  if(!robot.surfaceHasIndirectForceSensor(surfaceName))
  {
    mc_rtc::log::error_and_throw<std::runtime_error>(
        "[{}] Surface {} does not have a force sensor attached", name_, surfaceName);
  }

```